crates/oxc_parser/src/js/expression.rs

```diff
@@ -451,13 +451,17 @@ impl<'a> ParserImpl<'a> {
     ///     `SubstitutionTemplate`[?Yield, ?Await, ?Tagged]
     pub(crate) fn parse_template_literal(&mut self, tagged: bool) -> TemplateLiteral<'a> {
         let span = self.start_span();
-        let mut expressions = self.ast.vec();
-        let mut quasis = self.ast.vec();
-        match self.cur_kind() {
+
+        let (quasis, expressions) = match self.cur_kind() {
             Kind::NoSubstitutionTemplate => {
+                let mut quasis = self.ast.vec_with_capacity(1);
```

Comment:
Small thing: This would be slightly better:

```rs
let mut quasis = self.ast.vec1(self.parse_template_element(tagged));
```

(1 less bounds check)